docs/Client.md

```diff
@@ -0,0 +1,1011 @@
+* **opaque** `unknown` (optional) - This argument parameter is passed through to `ConnectData`
+
+#### Parameter: `ConnectData`
+
+* **statusCode** `number`
+* **headers** `http.IncomingHttpHeaders`
+* **socket** `stream.Duplex`
+* **opaque** `unknown`
+
+#### Example - Connect request with echo
+
+```js
+'use strict'
+const { createServer } = require('http')
+const { Client } = require('undici')
+
+const server = createServer((request, response) => {
+  throw Error('should never get here')
+})
+
+server.on('connect', (req, socket, firstBodyChunk) => {
+  socket.write('HTTP/1.1 200 Connection established\r\n\r\n')
+
+  let data = firstBodyChunk.toString()
+  socket.on('data', (buf) => {
+    data += buf.toString()
+  })
+
+  socket.on('end', () => {
+    socket.end(data)
+  })
+})
+
+server.listen(() => {
+  const client = new Client(`http://localhost:${server.address().port}`)
+
+  client
+    .connect({ path: '/' })
+    .then(({ socket }) => {
+      const wanted = 'Body'
+      let data = ''
+      socket.on('data', d => { data += d })
+      socket.on('end', () => {
+        console.log(`Data received: ${data.toString()} | Data wanted: ${wanted}`)
+        client.close()
+        server.close()
+      })
+      socket.write(wanted)
+      socket.end()
+    })
+})
+```
+
+### `Client.destroy([error][, callback])`
+
+Destroy the client abruptly with the given error. All the pending and running requests will be asynchronously aborted and error. Waits until socket is closed before invoking the callback (or returnning a promise if no callback is provided). Since this operation is asynchronously dispatched there might still be some progress on dispatched requests.
+
+Both arguments are optional; the method can be called in four different ways:
+
+```js
+client.destroy() // -> Promise
+client.destroy(new Error()) // -> Promise
+client.destroy(() => {}) // -> void
+client.destroy(new Error(), () => {}) // -> void
+```
+
+Arguments:
+
+* **error** `Error | null` (optional)
+* **callback** `() => void` (optional)
+
+Returns: `void | Promise<void>` - Only returns a `Promise` if no `callback` argument was passed
+
+#### Example - Request is aborted when Client is destroyed
+
+```js
+'use strict'
+const { createServer } = require('http')
+const { Client } = require('undici')
+
+const server = createServer((request, response) => {
+  response.end('undici')
+})
+
+server.listen(() => {
+  const client = new Client(`http://localhost:${server.address().port}`)
+
+  const request = client.request({
+    path: '/',
+    method: 'GET'
+  })
+
+  client.destroy()
+    .then(() => {
+      // Still waits for requests to complete
+      console.log('Client destroyed')
+      server.close()
+    })
+
+  // The request promise will reject with an Undici Client Destroyed error
+  request.catch(error => {
+    console.error(error)
+  })
+})
+```
+
+### `Client.dispatch(options, handlers)`
```

Comment:
It would be nice to put this in a separate section for library developers?

Reply:
Yeah I touched on this earlier in my replies, but to reiterate...

This documentation page is supposed to be a one stop place for all API information. For more generalized documentation - that should be captured in another file or in another, higher level section on this same page.